src/platform_impl/linux/x11/window.rs

```diff
@@ -1789,8 +1789,8 @@ impl UnownedWindow {
                         | xproto::EventMask::SUBSTRUCTURE_NOTIFY,
                 ),
                 [
-                    (window.x as u32 + xinput_fp1616_to_float(pointer.win_x) as u32),
-                    (window.y as u32 + xinput_fp1616_to_float(pointer.win_y) as u32),
+                    (window.x + xinput_fp1616_to_float(pointer.win_x) as i32) as u32,
+                    (window.y + xinput_fp1616_to_float(pointer.win_y) as i32) as u32,
```

Comment:
Given that you can repro that, what is _negative_ here? Since I don't see how it can be done otherwise? If it's negative, why the message is `u32`? I guess it's `i32` but encoded as `u32`?

Reply:
Yeah, this code doesn't make much sense to me. How does it work now when it didn't work before?

Reply:
It did overflow when doing math operation because something was negative, then it just was interpreted as `u32`, but `as` just changes interpretation, it doesn't really do operations which have overflow checking.